src/dependency_injector/providers.pyi

```diff
@@ -249,9 +250,9 @@ class Factory(Provider[T]):
     provided_type: Optional[Type]
     def __init__(self, provides: Optional[_Callable[..., T]] = None, *args: Injection, **kwargs: Injection) -> None: ...
     @property
-    def cls(self) -> T: ...
+    def cls(self) -> Type[T]: ...
```

Comment:
At the same time, ``Factory`` provider still has a strong use case when it is used with a class. When you need to access factory class, attribute ``.cls`` is better from the readability and understandability perspectives. Even despite ``.cls`` and ``.provides`` are now aliases and have the same implementation, I think we should treat them differently. With that said, I think that's ok if we type ``.cls -> Type[T]``, and ``.provides -> _Callable[..., T]``.

Reply:
okidoki 👍 

once we resolve the conversation at https://github.com/ets-labs/python-dependency-injector/pull/491/files#r692507266, I'll do the changes 🙏

Reply:
done 47cf4c2